### PR TITLE
output error message if we can't start a php server

### LINKED ISSUE #46: bin/http_test_server not functional on Windows
**PHP version**: 7.4.9

**Description**

For guzzle/guzzle#2741 I attempted to set up Windows CI. This did not work, because `bin/http_test_server` cannot be used on Windows as is.

**How to reproduce**

Within a Guzzle clone attempt to use `vendor/bin/http_test_server.bat`.

**Possible Solution**

The issue appears to be that `vendor/bin/http_test_server` is *not* a symlink on Windows. Instead the file looks something like this:

```
#!/usr/bin/env sh

dir=$(cd "${0%[/\\]*}" > /dev/null; cd "../php-http/client-integration-tests/bin" && pwd)

if [ -d /proc/cygdrive ]; then
    case $(which php) in
        $(readlink -n /proc/cygdrive)/*)
            # We are in Cygwin using Windows php, so the path must be translated
            dir=$(cygpath -m "$dir");
            ;;
    esac
fi

"${dir}/http_test_server" "$@"
```

As `${dir}/http_test_server` is executed and not `vendor/bin/http_test_server` as a symlink the `BASH_SOURCE` in `http_test_server` will point to `vendor/php-http/client-integration-tests/bin/http_test_server`, causing the script to search the autoloader in `vendor/php-http/client-integration-tests/bin/../autoload.php`:

https://github.com/TimWolla/guzzle/runs/1153859444?check_suite_focus=true

**Additional context**

From my patched GitHub Actions CI:

```
++ cd ./vendor/bin
++ cd ../php-http/client-integration-tests/bin
++ pwd
+ dir=/d/a/guzzle/guzzle/vendor/php-http/client-integration-tests/bin
+ '[' -d /proc/cygdrive ']'
+ case $(which php) in
++ which php
++ readlink -n /proc/cygdrive
+ /d/a/guzzle/guzzle/vendor/php-http/client-integration-tests/bin/http_test_server
+++ dirname /d/a/guzzle/guzzle/vendor/php-http/client-integration-tests/bin/http_test_server
++ cd /d/a/guzzle/guzzle/vendor/php-http/client-integration-tests/bin
++ pwd
+ DIR=/d/a/guzzle/guzzle/vendor/php-http/client-integration-tests/bin
+ '[' -f /d/a/guzzle/guzzle/vendor/php-http/client-integration-tests/bin/../autoload.php ']'
+ '[' -f /d/a/guzzle/guzzle/vendor/php-http/client-integration-tests/bin/../vendor/autoload.php ']'
```

### PR DESCRIPTION
**Pull request by dbu**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | fixes #46
| License         | MIT

Improve DX when we fail to start a web server

### CODE REVIEW

File: bin/http_test_server [hunk 1]
```diff
@@ -8,4 +8,8 @@ if [ -f "$DIR/../autoload.php" ]; then
 # Development
 elif [ -f "$DIR/../vendor/autoload.php" ]; then
     php -S 127.0.0.1:10000 -t "$DIR/../fixture"
+# Not working, e.g. windows which does not do symlinks
+else
+    echo "*** Can't find the fixture folder ***"
+    echo "Please write your own way to start a PHP web server on port 10000 for the 'fixture' directory."
 fi
```
Comment (TimWolla) [inline bin/http_test_server:15]:
```suggestion
else
    echo "*** Can't find the fixture folder ***" >&2
    echo "Please write your own way to start a PHP web server on port 10000 for the 'fixture' directory." >&2
    exit 1
fi
```

a) Print to STDERR.
b) Kill the script with an error status to stop CI from progressing.

**Review by TimWolla — CHANGES_REQUESTED** [review]

File: bin/http_test_server (hunk 1 above)
Comment (dbu) [reply bin/http_test_server]: indeed, both good points

**Review by TimWolla — APPROVED** [review]
Tested locally. Looks good to me now. Thanks!

**Comment by dbu** [comment]
thanks a lot for reporting this!

**Review by Nyholm — APPROVED** [review]
Thank you

**Comment by dbu** [comment]
the build of this package itself does look like it needs some love. but the change in this PR does not break that part, so i still merged it.

**Pull request merged**